crates/viewer/re_renderer/src/renderer/video.rs

```diff
@@ -0,0 +1,526 @@
+use re_video::VideoLoadError;
+
+use crate::resource_managers::GpuTexture2D;
+use crate::wgpu_resources::GpuTexturePool;
+use crate::wgpu_resources::TextureDesc;
+use crate::RenderContext;
+
+use re_video::TimeMs;
+
+pub struct Video {
+    decoder: decoder::VideoDecoder,
+}
+
+impl Video {
+    pub fn load(
+        render_context: &RenderContext,
+        media_type: Option<&str>,
+        data: &[u8],
+    ) -> Result<Self, VideoError> {
+        let data = match media_type {
+            Some("video/mp4") => re_video::load_mp4(data)?,
+            _ => return Err(VideoError::Load(VideoLoadError::UnknownMediaType)),
+        };
+        let decoder = VideoDecoder::new(render_context, data).ok_or_else(|| VideoError::Init)?;
+
+        Ok(Self { decoder })
+    }
+
+    pub fn duration_ms(&self) -> f64 {
+        self.decoder.duration_ms()
+    }
+
+    pub fn width(&self) -> u32 {
+        self.decoder.width()
+    }
+
+    pub fn height(&self) -> u32 {
+        self.decoder.height()
+    }
```

Comment:
We probably need to consider non-square pixels here, unless the web video decoder abstracts that away for us

Reply:
I think it does? The pixel aspect ratio seems to be codec-specific information, which we pass directly to the video decoder. It handles format conversion (e.g. YUV -> RGB), scaling, etc., stuff that we also likely won't have to worry about with ffmpeg/gstreamer

Reply:
DVD:s used non-square pixels, but I don't expect anyone to log a DVD video to Rerun.
Let's just punt on this until it becomes a problem, if ever.